src/scarlet2/scene.py

```diff
@@ -96,6 +102,10 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_value, traceback):
         Scenery.scene = None
+        if self.check_scene:
+            from .validation import check_scene
+
+            check_scene(self)
```

Comment:
It's an elegant place for an automatic check. The only reason I don't quite like it is that if somebody adds one source after the initial creation of the scene, all existing sources would be checked again.

Reply:
Also very reasonable. It occurred to me that we could add a couple members to the `Observation`, `Scene`, or `Source` classes like: `self.validation_status: [UNCHECKED | PASSED | FAILED]` as well as a `self.validation_errors: list[ValidationDataclass]`.

But it seems that the class hierarchy with Equinox is such that updating class instance values isn't allowed. (Or rather, you can append to a list, but you can't update the status after creating the class instance.

For now I'm going to just return a list of validation error dataclasses from `check_*`, where each element in the list will have context of where it originated.

Reply:
We've pulled the checking out of the `__exit__` method, and are focusing the checks on the sources at this point. The user has access to a `check_scene` function that will simply loop over the sources in the scene and check each one.